### PR TITLE
Add splashBorderRadius to TabBar

### PR DESCRIPTION
**Pull request by nayeemtby**
This PR is adding splashBorderRadius to TabBar that provides borderRadius to InkWell.

Addresses Feature Request #94290

Before: https://youtu.be/S2qBpaE2wOo
After: https://youtu.be/xvK_uZXVmDc

## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I read and followed the [Flutter Style Guide], including [Features we expect every widget to implement].
- [x] I signed the [CLA].
- [x] I listed at least one issue that this PR fixes in the description above.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] I added new tests to check the change I am making, or this PR is [test-exempt].
- [x] All existing and new tests are passing.

If you need help, consider asking for advice on the #hackers-new channel on [Discord].

[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[test-exempt]: https://github.com/flutter/flutter/wiki/Tree-hygiene#tests
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[Features we expect every widget to implement]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#features-we-expect-every-widget-to-implement
[CLA]: https://cla.developers.google.com/
[flutter/tests]: https://github.com/flutter/tests
[breaking change policy]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes
[Discord]: https://github.com/flutter/flutter/wiki/Chat

### CODE REVIEW

File: packages/flutter/lib/src/material/tabs.dart
Comment (werainkhatri) [inline packages/flutter/lib/src/material/tabs.dart:861]: This is repeated below.

```suggestion
```

File: packages/flutter/lib/src/material/tabs.dart
Comment (werainkhatri) [inline packages/flutter/lib/src/material/tabs.dart:863]:
```suggestion
  /// For example, to match the [BoxDecoration.borderRadius] provided as [indicator].
```

**Review by werainkhatri — CHANGES_REQUESTED** [review]
Thank you for your contribution. Just required a few doc fixes.

**Comment by nayeemtby** [comment]
> Thank you for your contribution. Just required a few doc fixes.

Welcome 😊, I don't have write access, so please add the 'waiting for tree to go green' label if everything is okay.

**Review by werainkhatri — APPROVED** [review]
32384589-a60f0e74-c078-11e7-9bc1-e5b5287aea9d

Although, you'll need an LGTM from another member (as per our policy). I'll ask for one on discord.

**Comment by nayeemtby** [comment]
> 32384589-a60f0e74-c078-11e7-9bc1-e5b5287aea9d
> 
> Although, you'll need an LGTM from another member (as per our policy). I'll ask for one on discord.

Thanks 😅

File: packages/flutter/lib/src/material/tabs.dart
Comment (chunhtai) [inline packages/flutter/lib/src/material/tabs.dart:860]:
```suggestion
  /// This can be useful if you want to match the [BoxDecoration.borderRadius] provided as [indicator].
```

**Review by chunhtai — APPROVED** [review]
LGTM, except one comment nit

File: packages/flutter/lib/src/material/tabs.dart
Comment (nayeemtby) [reply packages/flutter/lib/src/material/tabs.dart:860]: Should I also change 'provided as' to 'provided in' ?

Comment (nayeemtby) [reply packages/flutter/lib/src/material/tabs.dart:860]: And as per https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#use-the-passive-voice-recommend-do-not-require should I do 'This can be useful' ?

**Pull request merged**
